src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -379,6 +391,240 @@ void CascadiaSettings::_LoadDynamicProfiles()
+    // we are on a UI thread - and so cannot call a blocking operation like .get() - we use
+    // a mutex and condition variable
+    std::condition_variable cv;
+    std::mutex mtx;
+    Windows::Foundation::Collections::IVectorView<Windows::ApplicationModel::AppExtensions::AppExtension> extensions;
+
+    auto findAllExtensions = [&]() -> winrt::fire_and_forget {
+        co_await resume_background();
+        const auto localExtensions = catalog.FindAllAsync().get();
+
+        std::unique_lock<std::mutex> lock{ mtx };
+        extensions = localExtensions;
+        cv.notify_all();
+    };
+
+    std::unique_lock<std::mutex> lock{ mtx };
+    findAllExtensions();
+    cv.wait(lock);
+
+    for (const auto& ext : extensions)
+    {
+        // Only apply the stubs if the package name is not in ignored namespaces
+        if (ignoredNamespaces.find(ext.Package().DisplayName().c_str()) == ignoredNamespaces.end())
+        {
+            // Likewise, getting the public folder from an extension is an async operation
+            // So we use another mutex and condition variable
+            std::condition_variable cv2;
+            std::mutex mtx2;
+            Windows::Storage::StorageFolder foundFolder{ nullptr };
+
+            auto findPublicFolder = [&]() -> winrt::fire_and_forget {
+                co_await resume_background();
+                const auto localFolder = ext.GetPublicFolderAsync().get();
+                std::unique_lock<std::mutex> lock{ mtx2 };
+                foundFolder = localFolder;
+                cv2.notify_all();
+            };
+
+            std::unique_lock<std::mutex> lock2{ mtx2 };
+            findPublicFolder();
+            cv2.wait(lock2);
+
+            // the StorageFolder class has its own methods for obtaining the files within the folder
+            // however, all those methods are Async methods
+            // you may have noticed that we need to resort to clunky implementations for async operations
+            // (they're right above)
+            // so for now we will just take the folder path and access the files that way
+            auto path = winrt::to_string(foundFolder.Path());
+            path.append(FragmentsSubDirectory);
+            const auto jsonFiles = _AccumulateJsonFilesInDirectory(til::u8u16(path));
+
+            // Provide the package name as the source
+            _AddOrModifyProfiles(jsonFiles, ext.Package().DisplayName().c_str());
+        }
+    }
+}
+
+// Method Description:
+// - Helper function to apply json stubs in the local app data folder and the global program data folder
+// Arguments:
+// - The directory to find json files in
+// - The set of ignored namespaces
+void CascadiaSettings::_ApplyJsonStubsHelper(const std::wstring_view directory, const std::unordered_set<std::wstring>& ignoredNamespaces)
+{
+    const std::filesystem::path root{ wil::ExpandEnvironmentStringsW<std::wstring>(directory.data()) };

```

Comment:
you're expanding the strings in every function -- you should only have to do it once!